pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -1127,6 +1127,21 @@ func TestValidatePipelineParameterVariables_Success(t *testing.T) {
 				Name: "a-param-intended-to-be-array", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.myArray[*])"},
 			}},
 		}},
+	}, {
+		name: "valid string parameter variables in matrix include",
```

Comment:
we might want to add a similar test for matrix.params if it does not already exist:

```
		name: "valid string parameter variables in matrix params",
		params: []ParamSpec{{
			Name: "baz", Type: ParamTypeString,
		}},
		tasks: []PipelineTask{{
			Name:    "bar",
			TaskRef: &TaskRef{Name: "bar-task"},
			Matrix: &Matrix{
				Params: []Param{{
					Name: "a-param", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.baz)"}},
				}},
		}},
```

Reply:
There is an existing test for "parameters in matrix are strings".